code/script.js

```diff
@@ -229,97 +389,139 @@ const setSecret = () => {
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(`Yes, the champion has ${value} hair! Keep all champions with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(`No, the champion doesn't have ${value} hair! Remove all champions that have ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+     } else if(category === 'costume') {
+      if (keep) {
+        alert(`Yes, the champion has ${value} costume! Keep all champions with ${value} costume`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      } else {
+        alert(`No, the champion doesn't have ${value} costume! Remove all champions that have ${value} costume`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      }
+    }
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    
+    generateBoard()
+  }
 
-  // Invoke a function to redraw the board with the remaining people.
-}
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+// Player can confirm their guess when clicking the guess button, it then stores the value and invokes function to check their guess.
+const guess = (suspectCharacter) => {
+  
+  const confirmGuess = confirm(`Is this champion your guess?`)
+  if (confirmGuess)
+  checkMyGuess(suspectCharacter)
+  
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+// If you confirm your guess, this function is invoked to display either if it was correct or false.
+const checkMyGuess = (suspectCharacter) => {
+
+  if (suspectCharacter === secret.name) {
+  winOrLoseText.innerHTML = `Victory! You guessed on ${suspectCharacter} which is the right champion!`
+  victorySound.play()
+} else {
+  winOrLoseText.innerHTML = `Defeat! The right champion was ${secret.name}...`
+  defeatSound.play()
+}
+  
+  winOrLose.style.display = 'flex';
+  
 }
 
 // Invokes the start function when website is loaded
-start()
+start();
```

Comment:
Just realized sometimes you use ; and sometimes no semi-colons. I think it could be because of prettier. 

If I can recommend something it would be to decide what you feel is best for you, to use it or not and that goes for "or '. In prettier you can set what you want to use so that it formats the way you decide! 
Small detail but again makes your code look smooth.

Reply:
I use prettier and do never add semicolon myself so thank you for giving me the tips to set what my default should be :)

Reply:
Great! Then me and you are in the same boat. I dont like the semi colon in javascript. :D